File: HW/HW10/HW10.java
import java.util.List;
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.Queue;
public class HW10 {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> levelList = new ArrayList<>();
        if (root == null) {
            return levelList;
        }
        Queue<TreeNode> queue = new LinkedList<>();
        queue.add(root);
        while (!queue.isEmpty()) {
            List<Integer> currentLevel = new ArrayList<>();
            int size = queue.size();
            TreeNode tempNode;
            for (int i = 0; i < size; i++) {
                tempNode = queue.poll();
                currentLevel.add(tempNode.val);
                if (tempNode.left != null) {
                    queue.add(tempNode.left);
                }
                if (tempNode.right != null) {
                    queue.add(tempNode.right);
                }
            }
            levelList.add(currentLevel);
        }
        return levelList;
    }
}

File: HW/HW10/HW10.py
from typing import Optional, List
from queue import Queue
class HW10(object):
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        levelList = []
        if root is None:
            return levelList
        queue = Queue()
        queue.put(root)
        while not queue.empty():
            currentLevel = []
            size = queue.qsize()
            for _ in range(size):
                tempNode = queue.get()
                currentLevel.append(tempNode.val)
                if tempNode.left is not None:
                    queue.put(tempNode.left)
                if tempNode.right is not None:
                    queue.put(tempNode.right)
            levelList.append(currentLevel)
        return levelList